onnxruntime/python/tools/quantization/calibrate.py

```diff
@@ -58,24 +148,64 @@ def augment_graph(self):
         added_outputs = []
         tensors_to_calibrate = set()
+        value_info_set = {} # value info set with specific type 
+        tensor_initializer = set()
+
+        for init in model.graph.initializer:
+            tensor_initializer.add(init.name)
+
+        # Target all ops with specific type
+        if implicitly_quantize_all_ops:
+            type_list = [onnx_proto.TensorProto.FLOAT, onnx_proto.TensorProto.FLOAT16]
+            value_info_set = self.get_value_info_by_type(model.graph.value_info, type_list)
 
         for node in model.graph.node:
-            should_be_calibrate = ((node.op_type in self.calibrate_op_types) and
-                                   (node.name not in self.black_nodes)) or (node.name in self.white_nodes)
-            if should_be_calibrate:
-                tensors_to_calibrate.update(node.input)
-                tensors_to_calibrate.update(node.output)
 
+            ########################
+            ## TODO: handle subgraph
+            ########################
+            '''
+            if len(node.attribute) > 0 and node.attribute[0].g.node: 
+                print(node.name)
+                print(node.attribute[0].g)
+                print("\n")
+                # for value_info in node.attribute[0].g.value_info:
+                    # print(value_info)
+            '''
+
+            if implicitly_quantize_all_ops:
+                for i in node.input:
+                    if i in tensor_initializer:
+                        continue
+                    if i in value_info_set:
+                        tensors_to_calibrate.add(i)
+                for o in node.input:
+                    if o in tensor_initializer:
+                        continue
+                    if o in value_info_set:
+                        tensors_to_calibrate.add(o)
+            else:
+                should_be_calibrate = ((node.op_type in self.calibrate_op_types) and
+                                       (node.name not in self.black_nodes)) or (node.name in self.white_nodes)
+                if should_be_calibrate:
+                    tensors_to_calibrate.update(node.input)
+                    tensors_to_calibrate.update(node.output)
+
+
+            # following has the chance to cause "RuntimeError: Set changed size during iteration"
             for tensor in tensors_to_calibrate:
                 if tensor in model.graph.initializer:
                     tensors_to_calibrate.remove(tensor)
 
+        print("tensors_to_calibrate:")
+        print(tensors_to_calibrate)
+
         for tensor in tensors_to_calibrate:
             # Adding ReduceMin nodes
             reduce_min_name = tensor + '_ReduceMin'
             reduce_min_node = onnx.helper.make_node('ReduceMin', [tensor], [tensor + '_ReduceMin'],
                                                     reduce_min_name,
-                                                    keepdims=0)
+                                                    keepdims=1)
```

Comment:
does changing keepdims impact the previous use case?

Reply:
Since we want to augment all the float nodes to generate cal table. And it's possible that some of the nodes' input value are 0 which may cause error. 
In reduction_ops.cc, it mentions why:
"Can't reduce on dim with value of 0 if 'keepdims' is false. Invalid output shape would be produced. input_shape:{0}"
I will modify the script to avoid this error.